app/services/PaymentAPIService.scala

```diff
@@ -0,0 +1,81 @@
+package services
+
+import java.io.IOException
+
+import play.api.libs.json.{JsObject, JsValue, Json}
+import play.api.libs.ws.{WSClient, WSResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object PaymentAPIService {
+  case class Email(value: String)
+  object Email {
+    def fromResponse(resp: WSResponse): Either[Throwable, Option[Email]] =
+      if (resp.status == 200) {
+        Right((resp.json \ "email").validate[String].asOpt.map(Email.apply))
+      } else {
+        Left(new IOException(resp.toString))
+      }
+  }
+}
+
+class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
+
+  private val createPaymentPath = "/contribute/one-off/paypal/create-payment"
+  private val executePaymentPath = "/contribute/one-off/paypal/execute-payment"
+
+  def getCreatePaymentURL: String = paymentAPIUrl.concat(createPaymentPath)
+  def getExecutePaymentURL: String = paymentAPIUrl.concat(executePaymentPath)
```

Comment:
Why did you use `concat` here instead of string interpolation (like on line 61)?

Reply:
This is fixed in: 1981272